packages/patternfly-4/react-core/src/components/TextInput/examples/FocusTextInput.js

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { TextInput, Button } from '@patternfly/react-core';
+
+class FocusTextInput extends React.Component {
+  static title = 'TextInput Focus';
+
+  state = {
+    value: ''
+  };
+
+  constructor(props) {
+    super(props);
+    this.inputRef = React.createRef();
+  }
+
+  handleTextInputChange = value => {
+    this.setState({ value });
+  };
+
+  componentDidUpdate() {
+    this.inputRef && this.inputRef.current.focus();
+  }
+
+  render() {
+    const { value } = this.state;
+const FocusTextInput = React.forwardRef(({ value, type, onChange }, ref) => (
```

Comment:
Example of how forwardRef should be used.
```jsx

// src/components/TextInput/TextInput.js
class TextInput extends React.Component {
  render() {
   const { innerRef } = this.props;
   ...rest
  }
}

export default React.forwardRef((props, ref) => <TextInput innerRef={ref} {...props}  />);

// Example.js
import { TextInput } from '@patternfly/react-core';

class Example extends React.Component {
  inputRef = React.createRef();

  render() {
    return <TextInput ref={this.inputRef} />
  }
}
```

I left out much of the example and component implementation, but that should get the idea acress. React.forwardRef should not be called in render at the very least.

Reply:
Got it, thanks. I will be updating in the next hours.

Reply:
It looks OK but now I noticed that `TextInput` props are not shown.

It seems like `React.forwardRef` doesn't return a function like `class Component extends React.Component { ... }` and `const Component = (props) => { ... }` do.

It also looks like this [won't be fixed](https://github.com/facebook/react/issues/12453).
As a workaround I decided to use `react-is.isForwardRef` inside a function and render from the object the component passing `props` passed to the function.

Let me know what you think.